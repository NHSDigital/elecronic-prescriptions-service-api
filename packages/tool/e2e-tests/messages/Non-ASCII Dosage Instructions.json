{
  "resourceType": "Bundle",
  "id": "3",
  "identifier": {
    "system": "https://tools.ietf.org/html/rfc4122",
    "value": "7c18b624-3519-490b-bfd6-2fb80c98b4f5"
  },
  "type": "message",
  "entry": [
    {
      "fullUrl": "urn:uuid:aef77afb-7e3c-427a-8657-2c427f71a272",
      "resource": {
        "resourceType": "MessageHeader",
        "id": "3599c0e9-9292-413e-9270-9a1ef1ead99c",
        "eventCoding": {
          "system": "https://fhir.nhs.uk/CodeSystem/message-event",
          "code": "prescription-order",
          "display": "Prescription Order"
        },
        "sender": {
          "identifier": {
            "system": "https://fhir.nhs.uk/Id/ods-organization-code",
            "value": "RBA"
          },
          "reference": "urn:uuid:56166769-c1c4-4d07-afa8-132b5dfca666",
          "display": "RAZIA|ALI"
        },
        "source": {
          "endpoint": "https://directory.spineservices.nhs.uk/STU3/Organization/RBA"
        },
        "destination": [
          {
            "endpoint": "https://int.api.service.nhs.uk/electronic-prescriptions/$post-message",
            "receiver": {
              "identifier": {
                "system": "https://fhir.nhs.uk/Id/ods-organization-code",
                "value": "X26"
              }
            }
          }
        ],
        "focus": []
      }
    },
    {
      "fullUrl": "urn:uuid:78d3c2eb-009e-4ec8-a358-b042954aa9b2",
      "resource": {
        "resourceType": "Patient",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/nhs-number",
            "value": "9449306613"
          }
        ],
        "name": [
          {
            "use": "usual",
            "family": "ANDERTON",
            "given": [
              "CHADWICK",
              "BRIAR"
            ],
            "prefix": [
              "MR"
            ]
          }
        ],
        "gender": "male",
        "birthDate": "2009-02-15",
        "address": [
          {
            "use": "home",
            "line": [
              "EAST LODGE",
              "ROOKERY HILL",
              "ASHTEAD"
            ],
            "postalCode": "KT21 1JA"
          }
        ],
        "generalPractitioner": [
          {
            "identifier": {
              "system": "https://fhir.nhs.uk/Id/ods-organization-code",
              "value": "A83008"
            }
          }
        ]
      }
    },
    {
      "fullUrl": "urn:uuid:a8c85454-f8cb-498d-9629-78e2cb5fa47a",
      "resource": {
        "resourceType": "Practitioner",
        "id": "a8c85454-f8cb-498d-9629-78e2cb5fa47a",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/sds-user-id",
            "value": "7020134158"
          },
          {
            "system": "https://fhir.hl7.org.uk/Id/nmc-number",
            "value": "12A1B1AA"
          }
        ],
        "name": [
          {
            "text": "DR Thomas Edwards"
          }
        ]
      }
    },
    {
      "fullUrl": "urn:uuid:56166769-c1c4-4d07-afa8-132b5dfca666",
      "resource": {
        "resourceType": "PractitionerRole",
        "id": "56166769-c1c4-4d07-afa8-132b5dfca666",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/sds-role-profile-id",
            "value": "100102238986"
          }
        ],
        "practitioner": {
          "reference": "urn:uuid:a8c85454-f8cb-498d-9629-78e2cb5fa47a"
        },
        "organization": {
          "reference": "urn:uuid:3b4b03a5-52ba-4ba6-9b82-70350aa109d8"
        },
        "code": [
          {
            "coding": [
              {
                "system": "https://fhir.nhs.uk/CodeSystem/NHSDigital-SDS-JobRoleCode",
                "code": "S8000:G8000:R8000",
                "display": "Clinical Practitioner Access Role"
              }
            ]
          }
        ],
        "telecom": [
          {
            "system": "phone",
            "value": "0123456791",
            "use": "work"
          }
        ]
      }
    },
    {
      "fullUrl": "urn:uuid:51793ac0-112f-46c7-a891-9af8cefb206e",
      "resource": {
        "resourceType": "CommunicationRequest",
        "status": "unknown",
        "subject": {
          "reference": "urn:uuid:78d3c2eb-009e-4ec8-a358-b042954aa9b2"
        },
        "payload": [
          {
            "contentString": "TEST PRESCRIPTION - DO NOT DISPENSE"
          }
        ],
        "requester": {
          "type": "Organization",
          "identifier": {
            "system": "https://fhir.nhs.uk/Id/ods-organization-code",
            "value": "RBA"
          }
        },
        "recipient": [
          {
            "type": "Patient",
            "identifier": {
              "system": "https://fhir.nhs.uk/Id/nhs-number",
              "value": "9449306613"
            }
          }
        ]
      }
    },
    {
      "fullUrl": "urn:uuid:f8542de8-9bad-45cd-9d8a-11e8c3f1c263",
      "resource": {
        "resourceType": "MedicationRequest",
        "id": "f8542de8-9bad-45cd-9d8a-11e8c3f1c263",
        "extension": [
          {
            "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionType",
            "valueCoding": {
              "system": "https://fhir.nhs.uk/CodeSystem/prescription-type",
              "code": "0104",
              "display": "Primary Care Prescriber - Nurse Independent/Supplementary prescriber"
            }
          }
        ],
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/prescription-order-item-number",
            "value": "dbfec92b-edb4-4c16-9fab-bcc2ebadb620"
          }
        ],
        "status": "active",
        "intent": "order",
        "category": [
          {
            "coding": [
              {
                "system": "http://terminology.hl7.org/CodeSystem/medicationrequest-category",
                "code": "outpatient",
                "display": "Outpatient"
              }
            ]
          }
        ],
        "medicationCodeableConcept": {
          "coding": [
            {
              "system": "http://snomed.info/sct",
              "code": "99999999",
              "display": "EpiPen Jr. 150micrograms/0.3ml (1 in 2,000) solution for injection auto-injectors (Viatris UK Healthcare Ltd)"
            }
          ]
        },
        "subject": {
          "reference": "urn:uuid:78d3c2eb-009e-4ec8-a358-b042954aa9b2"
        },
        "authoredOn": "2021-05-07T14:47:29+00:00",
        "requester": {
          "reference": "urn:uuid:56166769-c1c4-4d07-afa8-132b5dfca666"
        },
        "groupIdentifier": {
          "extension": [
            {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionId",
              "valueIdentifier": {
                "system": "https://fhir.nhs.uk/Id/prescription",
                "value": "2809fb3d-9eda-4da7-afb4-a0febd345da7"
              }
            }
          ],
          "system": "https://fhir.nhs.uk/Id/prescription-order-number",
          "value": "E8A565-A83003-F1D9BO"
        },
        "courseOfTherapyType": {
          "coding": [
            {
              "system": "http://terminology.hl7.org/CodeSystem/medicationrequest-course-of-therapy",
              "code": "acute"
            }
          ]
        },
        "dosageInstruction": [
          {
            "text": "¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿƒΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩαβγδεζηθικλμνξοπρςστυφχψωϑϒϖ•…′″‾⁄℘ℑℜ™ℵ←↑→↓↔↵⇐⇑⇒⇓⇔∀∂∃∅∇∈∉∋∏∑−∗√∝∞∠∧∨∩∪∫∴∼≅≈≠≡≤≥⊂⊃⊄⊆⊇⊕⊗⊥⋅⌈⌉⌊⌋⟨⟩◊♠♣♥♦\"&<>ŒœŠšŸˆ˜–—‘’‚“”„†‡‰‹›€"
          }
        ],
        "dispenseRequest": {
          "extension": [
            {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-PerformerSiteType",
              "valueCoding": {
                "system": "https://fhir.nhs.uk/CodeSystem/dispensing-site-preference",
                "code": "P1"
              }
            }
          ],
          "performer": {
            "identifier": {
              "system": "https://fhir.nhs.uk/Id/ods-organization-code",
              "value": "VNCEL"
            }
          },
          "quantity": {
            "value": 1,
            "unit": "pre-filled disposable injection",
            "system": "http://snomed.info/sct",
            "code": 11111111
          },
          "expectedSupplyDuration": {
            "value": 28,
            "unit": "day",
            "system": "http://unitsofmeasure.org",
            "code": "d"
          }
        },
        "substitution": {
          "allowedBoolean": false
        },
        "note": []
      }
    },
    {
      "fullUrl": "urn:uuid:3b4b03a5-52ba-4ba6-9b82-70350aa109d8",
      "resource": {
        "resourceType": "Organization",
        "identifier": [
          {
            "system": "https://fhir.nhs.uk/Id/ods-organization-code",
            "value": "A83003"
          }
        ],
        "type": [
          {
            "coding": [
              {
                "system": "https://fhir.nhs.uk/CodeSystem/organisation-role",
                "code": "76",
                "display": "GP PRACTICE"
              }
            ]
          }
        ],
        "name": "HALLGARTH SURGERY",
        "address": [
          {
            "use": "work",
            "type": "both",
            "line": [
              "HALLGARTH SURGERY, CHEAPSIDE"
            ],
            "city": "SHILDON",
            "district": "COUNTY DURHAM",
            "postalCode": "DL4 2HP"
          }
        ],
        "telecom": [
          {
            "system": "phone",
            "value": "01823 333444",
            "use": "work"
          }
        ],
        "partOf": {
          "identifier": {
            "system": "https://fhir.nhs.uk/Id/ods-organization-code",
            "value": "84H"
          },
          "display": "NHS COUNTY DURHAM CCG -NHS NORTH EAST AND NORTH CUMBRIA ICB "
        }
      }
    }
  ]
}
