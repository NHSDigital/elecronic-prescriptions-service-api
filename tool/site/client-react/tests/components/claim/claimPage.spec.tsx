import {render, waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import ClaimPage from "../../../src/components/claim/claimPage"
import userEvent from "@testing-library/user-event"
import {readMessage} from "./messages/messages"

const baseUrl = "baseUrl/"
const prescriptionId = "7A9089-A83008-56A03J"

const prescriptionOrderUrl = `${baseUrl}prescription/${prescriptionId}`
const dispenseNotificationUrl = `${baseUrl}dispense/history?prescription_id=${prescriptionId}`
const claimUrl = `${baseUrl}dispense/claim`

const prescriptionOrder = readMessage("prescriptionOrder.json")
const dispenseNotification = readMessage("dispenseNotification.json")

beforeEach(() => moxios.install())

afterEach(() => moxios.uninstall())

test("Displays loading text while prescription data is being requested", async () => {
  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Retrieving prescription details."))

  expect(screen.getByText("Loading...")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays claim form if prescription details are retrieved successfully", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: {
      dispense_notifications: [dispenseNotification]
    }
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Claim for Dispensed Medication"))

  expect(screen.getByText("Claim")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays an error if prescription-order not found", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: null
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Error"))

  expect(screen.getByText("Prescription order not found. Is the ID correct?")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays an error if dispense-notification not found", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: {
      dispense_notifications: []
    }
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Error"))

  expect(screen.getByText("Dispense notification not found. Has this prescription been dispensed?")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays an error on invalid response", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: {}
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: {
      dispense_notifications: [{}]
    }
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Error"))

  expect(screen.getByText("Failed to retrieve prescription details.")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays loading text while claim is being submitted", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: {
      dispense_notifications: [dispenseNotification]
    }
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Claim for Dispensed Medication"))
  userEvent.click(screen.getByText("Claim"))
  await waitFor(() => screen.getByText("Loading..."))

  expect(screen.getByText("Sending claim.")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays claim result", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: {
      dispense_notifications: [dispenseNotification]
    }
  })
  moxios.stubRequest(claimUrl, {
    status: 200,
    response: "Mock result"
  })

  const {container} = render(<ClaimPage baseUrl={baseUrl} prescriptionId={prescriptionId}/>)
  await waitFor(() => screen.getByText("Claim for Dispensed Medication"))
  userEvent.click(screen.getByText("Claim"))
  await waitFor(() => screen.getByText("Result"))

  expect(screen.getByText(JSON.stringify("Mock result"))).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})
