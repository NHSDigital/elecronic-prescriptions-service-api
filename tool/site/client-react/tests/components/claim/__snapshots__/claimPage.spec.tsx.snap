// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays an error if dispense-notification not found 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Error</label></h1><span class=\\"nhsuk-error-message\\" role=\\"alert\\"><span class=\\"nhsuk-u-visually-hidden\\">Error: </span>Dispense notification not found. Has this prescription been dispensed?</span>"`;

exports[`Displays an error if prescription-order not found 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Error</label></h1><span class=\\"nhsuk-error-message\\" role=\\"alert\\"><span class=\\"nhsuk-u-visually-hidden\\">Error: </span>Prescription order not found. Is the ID correct?</span>"`;

exports[`Displays an error on invalid response 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Error</label></h1><span class=\\"nhsuk-error-message\\" role=\\"alert\\"><span class=\\"nhsuk-u-visually-hidden\\">Error: </span>Failed to retrieve prescription details.</span>"`;

exports[`Displays claim form if prescription details are retrieved successfully 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Claim for Dispensed Medication</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Methotrexate 10mg/0.2ml solution for injection pre-filled syringes</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">Item fully dispensed</dd>
        </div>
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Quantity Dispensed</dt>
          <dd class=\\"nhsuk-summary-list__value\\">63 tablet</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\">
        <div class=\\"nhsuk-checkboxes\\" id=\\"products.0.patientPaid.boxes\\">
          <div class=\\"nhsuk-checkboxes__item\\"><input class=\\"nhsuk-checkboxes__input\\" name=\\"products.0.patientPaid\\" id=\\"products.0.patientPaid.box\\" type=\\"checkbox\\" value=\\"false\\"><label class=\\"nhsuk-label nhsuk-checkboxes__label\\" id=\\"products.0.patientPaid.box--label\\" for=\\"products.0.patientPaid.box\\">Patient Paid</label></div>
        </div>
      </div>
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"button\\">Add Endorsement</button></div>
    </fieldset>
  </div>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Prescription Charge Exemption</legend>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"exemption.code--label\\" for=\\"exemption.code\\">Exemption Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"exemption.code--label\\" name=\\"exemption.code\\" id=\\"exemption.code\\">
          <option value=\\"0001\\">Patient has paid appropriate charges</option>
          <option value=\\"0002\\">is under 16 years of age</option>
          <option value=\\"0003\\">is 16, 17 or 18 and in full-time education</option>
          <option value=\\"0004\\">is 60 years of age or over</option>
          <option value=\\"0005\\">has a valid maternity exemption certificate</option>
          <option value=\\"0006\\">has a valid medical exemption certificate</option>
          <option value=\\"0007\\">has a valid prescription pre-payment certificate</option>
          <option value=\\"0008\\">has a War Pension exemption certificate</option>
          <option value=\\"0009\\">is named on a current HC2 charges certificate</option>
          <option value=\\"0010\\">was prescribed free-of-charge contraceptives</option>
          <option value=\\"0011\\">gets income support (IS)</option>
          <option value=\\"0012\\">gets income based Job Seeker's Allowance (JSA (IB))</option>
          <option value=\\"0013\\">is entitled to, or named on a VALID NHS tax credit exemption certificate</option>
          <option value=\\"0014\\">has a partner who gets Pension Credit Guarantee Credit (PGCC)</option>
          <option value=\\"0015\\">Patient does not need to pay the prescription charge</option>
        </select></div>
      <div class=\\"nhsuk-form-group\\">
        <div class=\\"nhsuk-checkboxes\\" id=\\"exemption.evidenceSeen.boxes\\">
          <div class=\\"nhsuk-checkboxes__item\\"><input class=\\"nhsuk-checkboxes__input\\" name=\\"exemption.evidenceSeen\\" id=\\"exemption.evidenceSeen.box\\" type=\\"checkbox\\" value=\\"false\\"><label class=\\"nhsuk-label nhsuk-checkboxes__label\\" id=\\"exemption.evidenceSeen.box--label\\" for=\\"exemption.evidenceSeen.box\\">Evidence Seen</label></div>
        </div>
      </div>
    </fieldset>
  </div>
  <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Claim</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"reset\\">Reset</button></div>
</form>"
`;

exports[`Displays claim result 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Claim Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Request%22\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">\\"JSON Request\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Request\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">XML Request</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Response%22\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">\\"JSON Response\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Response\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">XML Response</pre>
    </div>
  </details>
</div>
<div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><a class=\\"nhsuk-button nhsuk-button--secondary\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"baseUrl/\\" type=\\"button\\">Back</a></div>"
`;

exports[`Displays loading text while claim is being submitted 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Loading...</label></h1><label class=\\"nhsuk-label\\">Sending claim.</label>"`;

exports[`Displays loading text while prescription data is being requested 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Loading...</label></h1><label class=\\"nhsuk-label\\">Retrieving prescription details.</label>"`;
